tests/client_generators/dependencies/test_async_base_client.py

```diff
@@ -67,6 +68,39 @@ class TestModel2(BaseModel):
     }
 
 
+@pytest.mark.asyncio
+async def test_execute_correctly_parses_top_level_list_variables(mocker):
+    class TestModel1(BaseModel):
+        a: int
+
+    fake_client = mocker.AsyncMock()
+    client = AsyncBaseClient(url="base_url", http_client=fake_client)
+    query_str = """
+    query Abc($v1: [[TestModel1!]!]!) {
+        abc(v1: $v1){
+            field1
+        }
+    }
+    """
+
+    await client.execute(
+        query_str,
+        {
+            "v1": [[TestModel1(a=1), TestModel1(a=2)]],
+        },
+    )
+
+    assert fake_client.post.called
+    assert len(fake_client.post.mock_calls) == 1
+    call_kwargs = fake_client.post.mock_calls[0].kwargs
+    assert call_kwargs["url"] == "base_url"
+    assert call_kwargs["json"] == {
+        "query": query_str,
+        "variables": {"v1": [[{"a": 1}, {"a": 2}]]},
+    }
+    assert not research(call_kwargs["json"]["variables"], lambda p, k, v: isinstance(v, BaseModel))
```

Comment:
This still has a dependency on boltons. Do you object to that being a dev dependency?

Reply:
Yes, we should have custom utility instead.

Reply:
I've replaced this with a check for the specific list structure we have here. It only exists in 2 places. I figured we can write something generic if we end up with more tests that require it.